src/Component/Filter/ComparisonFilter/ComparisonFilterUi.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+
+import { Row, Col } from 'antd';
+
+import AttributeCombo from '../AttributeCombo/AttributeCombo';
+import OperatorCombo from '../OperatorCombo/OperatorCombo';
+import TextFilterField from '../TextFilterField/TextFilterField';
+import NumberFilterField from '../NumberFilterField/NumberFilterField';
+
+import { ComparisonFilter } from 'geostyler-style';
+
+import './ComparisonFilterUi.css';
+
+/**
+ * UI for a ComparisonFilter consisting of
+ *
+ *   - A combo to select the attribute
+ *   - A combo to select the operator
+ *   - An input field for the value
+ */
+class ComparisonFilterUi extends React.Component<any, any> {
+
+  // PROPS
+
+  /** the internal data object (definition structure) */
+  internalDataDef: any;
+
+  /** placeholder shown when combobox has no selection */
+  placeholder: string = 'Enter Value';
+
+  /** Handler function which is executed when the filter values (thus the filter itself) changes */
+  onFilterChange: Function;
+
+  // NON PROPS / MEMBERS
+
+  /** */
+  attribute: string;
+
+  /** */
+  attributeType: string;
+
+  /** TODO replace with appropriate type once it is available */
+  operator: '==' | '*=' | '!=' | '<' | '<=' | '>' | '>=';
+
+  /** TODO replace with appropriate type once it is available */
+  value: string | number | boolean;
+
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      textFieldVisible: true,
+      numberFieldVisible: false
+    };
+  }
+
+  /**
+   * Handler function, which is executed, when to underlying filter attribute changes.
+   *
+   * Changes the input field for the filter value and stores the appropriate attribute name as member.
+   */
+  onAttributeChange = (newAttrName: string) => {
```

Comment:
Please transform methods to following syntax and use `.bind(this)` if needed:

```tsx
onAttributeChange(newAttrName: string) {
```

Otherwise styleguidist doesn't recognize this methods as methods but as properties.

Reply:
I will check that

Reply:
I wasn't aware that TypeScript recommends this way of declaring class methods. So 
we should stick to the TypeScript recommendation for know.